diff --git a/node_modules/@types/react-datepicker/index.d.ts b/node_modules/@types/react-datepicker/index.d.ts
index d24f35c..4f4ecb4 100755
--- a/node_modules/@types/react-datepicker/index.d.ts
+++ b/node_modules/@types/react-datepicker/index.d.ts
@@ -35,6 +35,22 @@ interface HighlightDates {
     [className: string]: Date[];
 }
 
+export interface ReactDatePickerCustomHeaderProps {
+    monthDate: Date;
+    date: Date;
+    changeYear(year: number): void;
+    changeMonth(month: number): void;
+    customHeaderCount: number;
+    decreaseMonth(): void;
+    increaseMonth(): void;
+    prevMonthButtonDisabled: boolean;
+    nextMonthButtonDisabled: boolean;
+    decreaseYear(): void;
+    increaseYear(): void;
+    prevYearButtonDisabled: boolean;
+    nextYearButtonDisabled: boolean;
+}
+
 export interface ReactDatePickerProps<CustomModifierNames extends string = never> {
     adjustDateOnChange?: boolean | undefined;
     allowSameDay?: boolean | undefined;
@@ -129,21 +145,7 @@ export interface ReactDatePickerProps<CustomModifierNames extends string = never
     previousMonthButtonLabel?: string | React.ReactNode | undefined;
     previousYearButtonLabel?: string | undefined;
     readOnly?: boolean | undefined;
-    renderCustomHeader?(params: {
-        monthDate: Date;
-        date: Date;
-        changeYear(year: number): void;
-        changeMonth(month: number): void;
-        customHeaderCount: number;
-        decreaseMonth(): void;
-        increaseMonth(): void;
-        prevMonthButtonDisabled: boolean;
-        nextMonthButtonDisabled: boolean;
-        decreaseYear(): void;
-        increaseYear(): void;
-        prevYearButtonDisabled: boolean;
-        nextYearButtonDisabled: boolean;
-    }): React.ReactNode;
+    renderCustomHeader?(params: ReactDatePickerCustomHeaderProps): React.ReactNode;
     renderDayContents?(dayOfMonth: number, date?: Date): React.ReactNode;
     required?: boolean | undefined;
     scrollableMonthYearDropdown?: boolean | undefined;
